src/python/pants/engine/internals/graph.py

```diff
@@ -693,39 +737,25 @@ async def resolve_dependencies(
         )
 
     literal_addresses = await MultiGet(Get(Address, AddressInput, ai) for ai in provided.addresses)
-    literal_ignored_addresses = await MultiGet(
-        Get(Address, AddressInput, ai) for ai in provided.ignored_addresses
+    literal_ignored_addresses = set(
+        await MultiGet(Get(Address, AddressInput, ai) for ai in provided.ignored_addresses)
     )
 
-    base_addresses = set()
-    all_generated_addresses = set()
-    used_ignored_addresses = set()
-    for addr in (
+    addresses: Set[Address] = set()
+    used_ignored_addresses: Set[Address] = set()
+    for addr in [
+        *subtarget_addresses,
         *literal_addresses,
         *itertools.chain.from_iterable(injected),
         *itertools.chain.from_iterable(inferred),
-    ):
+    ]:
         if addr in literal_ignored_addresses:
             used_ignored_addresses.add(addr)
         else:
-            if addr.is_base_target:
-                base_addresses.add(addr)
-            else:
-                all_generated_addresses.add(addr)
-
-    # We check if a generated subtarget's original base target is already included or if ts base
-    # target is the target that we're resolving dependencies for. In either of these cases, it
-    # would be redundant to include the generated subtarget.
```

Comment:
Why remove?

Reply:
Same as https://github.com/pantsbuild/pants/pull/10511#discussion_r466061065 ... have updated the tests for this now (not pushed yet).